client/interfaces.go

```diff
@@ -82,6 +88,20 @@ func (client *Client) Interfaces() (interfaces Interfaces, err error) {
 	return
 }
 
+// InterfaceIndex returns the list of all interface names.
+//
+// The result will only have three fields populated, Name, Summary and Used.
+func (client *Client) InterfaceIndex() (ifaces []Interface, err error) {
```

Comment:
```
/v2/snaps
/v2/interfaces
/v2/changes
/v2/users
/v2/sections
/v2/aliases
```

@zyga @chipaca  Something smelling strange about /v2/interface. We should try to preserve some sort of consistency in the API across the several endpoints.

Looking through it, here is a proposal:

1. Rename the current Interfaces method to Connections (we can worry about the type later)
2. Use `Interfaces` to return []*Interface; note the pointer result. Every type should be a pointer in these cases unless you have a good reason to believe return by value is the right thing to do (again, see existing cases)
3. Introduce `InterfacesOptions` as `{Doc, Plugs, Slots, Connected bool}`, meaning whether to return documentation, plugs, slots, and whether to return only connected interfaces.
4. We can provide "connected" via a "select" field, analogous to the one we use for snaps, and then allow specifying "select=all" so that we preserve the result as-is when "select" is not provided, preserving compatibility with existing clients (including the current Interfaces method which will now become Connections). Eventually we can obsolete that result as we'll likely kill the "interfaces" command and introduce a more polished "connections" one, per our conversations.
5. We should also support a "names" parameter in the Interfaces method analogous to the List method ("interfaces" parameter in the HTTP request, if we want to preserve the convention established by snaps). This would allow only a subselection of interfaces to be listed.

What do you think?